config/peerpods/podvm/aws-podvm-image-handler.sh

```diff
@@ -0,0 +1,342 @@
+#!/bin/bash
+# FILEPATH: aws-podvm-image-handler.sh
+
+# This script is used to create or delete AWS AMI for podvm
+# The basic assumption is that the required variables are set as environment variables in the pod
+# Typically the variables are read from configmaps and set as environment variables in the pod
+# The script will be called with one of the following options:
+# Create image (-c)
+# Delete image (-C)
+
+set -x
+# include common functions from lib.sh
+# shellcheck source=/dev/null
+# The directory is where aws-podvm-image-handler.sh is located
+source "$(dirname "$0")"/lib.sh
+
+# Function to verify that the required variables are set
+
+function verify_vars() {
+    # Ensure CLOUD_PROVIDER is set to aws
+    [[ -z "${CLOUD_PROVIDER}" || "${CLOUD_PROVIDER}" != "aws" ]] && error_exit "CLOUD_PROVIDER is empty or not set to aws"
+
+    [[ -z "${AWS_ACCESS_KEY_ID}" ]] && error_exit "AWS_ACCESS_KEY_ID is not set"
+    [[ -z "${AWS_SECRET_ACCESS_KEY}" ]] && error_exit "AWS_SECRET_ACCESS_KEY is not set"
+
+    # Packer variables
+    [[ -z "${INSTANCE_TYPE}" ]] && error_exit "INSTANCE_TYPE is not set"
+    [[ -z "${PODVM_DISTRO}" ]] && error_exit "PODVM_DISTRO is not set"
+
+    # if AWS_REGION is empty, try to get it from the metadata service
+    if [[ ! "${AWS_REGION}" ]]; then
+        AWS_REGION=$(curl -m 30 -s --show-error http://169.254.169.254/latest/meta-data/placement/region ||
+            error_exit "Failed to get AWS_REGION from metadata service")
```

Comment:
actually if we do this here we can remove the validation of it from the code, however, I'm not sure we should :P

Reply:
Yeah, let the validation be there in the code as it will help when we add other image generation methods.